### ISSUE TITLE
no-switch-case-fall-through does not detect conditionals

### ISSUE THREAD

**Issue by domoritz** [palantir/tslint#950]
tslint wants me to add a break but ts doesn't allow it. 

``` ts
switch (x) {
  case a:
    if (foo) {
      break;
    } else {
      break;
    } 
    // tslint expects a break here but ts doesn't allow it because it is unreachable code
  case b:
    break;
}
```

**Comment by joskuijpers** [comment]
+1

**Comment by ajafff** [comment]
#1815 brought me here

I would like to fix this issue.
But before I do, I'd like to know if this rule is still necessary and supported, since the compiler also has the ability to check for switch-case-fallthrough.
So is this worth the effort or will this rule be deprecated soon?

**Comment by domoritz** [comment]
I think it is still relevant because the compiler check is implemented in a way that it is less flexible. But I suggest that you find a few test cases and see whether they raise warnings/errors in tslint vs tsc.
